mlflow/tracing/types/wrapper.py

```diff
@@ -166,19 +180,9 @@ def end(self):
         if self.status.status_code != TraceStatus.ERROR:
             self.set_status(SpanStatus(TraceStatus.OK))
 
-        with self._span._lock:
-            if self._span._start_time is None:
-                _logger.warning("Calling end() on a not started span. Ignoring.")
-                return
-            if self._span._end_time is not None:
-                _logger.warning("Calling end() on an ended span. Ignoring.")
-                return
-
-            self._span._end_time = time_ns()
-
-        self._span._span_processor.on_end(self)
+        self._span.end()
```

Comment:
Why don't we need to lock this?

Reply:
The end() should always be called by one thread.